digiwf-integrations/digiwf-dms-integration/digiwf-dms-integration-core/src/main/java/de/muenchen/oss/digiwf/dms/integration/adapter/out/s3/S3Adapter.java

```diff
@@ -53,33 +55,87 @@ public List<Content> loadFiles(final List<String> filepaths, final String fileCo
         return contents;
     }
 
-    private List<Content> getFilesFromFolder(String folderpath, final String domainSpecificS3Storage) {
+    private void validateFileSizes(final List<String> filePaths, final String fileContext, final String s3Storage) {
+        // Collect file sizes along with their paths
+        final Map<String, Long> fileSizesWithPaths = getFileSizesWithPaths(filePaths, fileContext, s3Storage);
+
+        // Filter files exceeding maximum size
+        final Map<String, Long> oversizedFiles = fileService.getOversizedFiles(fileSizesWithPaths);
+        // Handle oversized files
+        if (!oversizedFiles.isEmpty()) {
+            final String filesOverMaxString = oversizedFiles.entrySet().stream()
+                    .map(entry -> entry.getKey() + ": " + DataSize.ofBytes(entry.getValue()).toMegabytes() + " MB")
+                    .collect(Collectors.joining(System.lineSeparator()));
+            throw new BpmnError(FILE_SIZE_ERROR,
+                    String.format("The following files exceed the maximum size of %d MB:%n%s", fileService.getMaxFileSize().toMegabytes(), filesOverMaxString));
```

Comment:
Größe in MB wird aufgerundet. Bei geringer Überschreitung wird zweimal die gleiche Zahl in der Meldung angezeigt.

Reply:
Ich habe mir angeschaut, wie man das berechnen könnte.

`String formattedSize = String.format("%.2f MB", sizeInMegabytes / 1024.0);`

Da die MaxFileSize aber auf 100 MB eingestellt wird, ist es eher unwahrscheinlich, dass die genau mit Nachkommastellen getroffen wird. In den wenigen Situationen, wo es auftreten kann, ist es aber egal.